crates/vm/levm/src/vm.rs

```diff
@@ -268,10 +268,7 @@ impl VM {
         }
     }
 
-    pub fn execute(
-        &mut self,
-        current_call_frame: &mut CallFrame,
-    ) -> Result<OpcodeSuccess, VMError> {
+    pub fn execute(&mut self, current_call_frame: &mut CallFrame) -> TransactionReport {
```

Comment:
Shouldn't `transact` return a `TransactionReport` instead?

Reply:
I've just seen that `transact` is indeed returning a `TransactionReport`, but do you think `execute` should be returning the same?

Reply:
Yeah it may not be the most accurate representation but it makes it more simple to use the same, the attributes we need in both scenarios are almost identical.

Maybe the other option would be to have another struct for that purpose (`ContextResult?`) and then on `transact` we should `execute` the VM and generate the `TransactionReport` based on `ContextResult` and the final state of the VM. It doesn't sound like a bad idea.

Let me know what you think.

Reply:
We leave this change for another PR.